File: backend/core/src/main/java/com/feed_grabber/core/config/S3Config.java
package com.feed_grabber.core.config;

import com.amazonaws.auth.AWSCredentials;
import com.amazonaws.auth.AWSStaticCredentialsProvider;
import com.amazonaws.auth.BasicAWSCredentials;
import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.AmazonS3Client;
import com.amazonaws.services.s3.AmazonS3ClientBuilder;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class S3Config {
    @Value("${amazon.accessKey}")
    String accessKey;
    @Value("${amazon.secretKey}")
    String accessSecret;

    @Bean
    public AmazonS3 generateS3Client() {
        AWSCredentials credentials = new BasicAWSCredentials(accessKey,accessSecret);
        return AmazonS3ClientBuilder
                .standard()
                .withCredentials(new AWSStaticCredentialsProvider(credentials)).build();
    }
}

File: backend/event_processor/src/main/kotlin/com/feed_grabber/event_processor/fileStorage/S3Config.kt
package com.feed_grabber.event_processor.fileStorage

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import software.amazon.awssdk.auth.credentials.AwsBasicCredentials
import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider
import software.amazon.awssdk.regions.Region
import software.amazon.awssdk.services.s3.S3Client
import java.net.URI


@Configuration
class InfrastructureConfiguration() {

    @Value("\${amazon.accessKey}")
    private lateinit var ACCESS_KEY: String

    @Value("\${amazon.secretKey}")
    private lateinit var SECRET_KEY: String

    @Value("\${amazon.endpointUrl}")
    private lateinit var ENDPOINT_URL: String

    private val region: String = "eu-central-1"

    @Bean
    fun s3Client(): S3Client {
        val credentials = AwsBasicCredentials.create(ACCESS_KEY, SECRET_KEY)
        val credentialsProvider = StaticCredentialsProvider.create(credentials)
        return S3Client.builder()
                .credentialsProvider(credentialsProvider)
                .endpointOverride(URI.create(ENDPOINT_URL))
                .region(Region.of(region))
                .build()
    }
}